pkg/producer/kafka.go

```diff
@@ -47,21 +46,20 @@ type ProducerInput struct {
 	KafkaLogErrors       bool
 	KafkaLogSuccesses    bool
 	EnableSaramaDebugLog bool
+	ProtoSchemaConvertor convertor.IPFIXToKafkaConvertor
 }
 
 type KafkaProducer struct {
-	producer             sarama.AsyncProducer
-	input                ProducerInput
-	protoSchemaConvertor convertor.IPFIXToKafkaConvertor
+	producer sarama.AsyncProducer
+	input    ProducerInput
 	// closeCh is required to exit the go routine that captures error messages from
 	// sarama go client.
 	closeCh chan struct{}
 }
 
 func NewKafkaProducer(input ProducerInput) *KafkaProducer {
 	return &KafkaProducer{
-		input:                input,
-		protoSchemaConvertor: convertor.ProtoSchemaConvertor[input.KafkaProtoSchema](),
+		input: input,
```

Comment:
Not from this PR, do we need to validate whether protoSchemaConvertor is nil since we are using it directly when publishing record?
Also, for other required field like version, it's better to check whether it is assigned with value otherwise the producer will not work well.

Reply:
Yeah better to have another check.. user is going to have a check for supported schemas anyway when fetching the protoschema convertors. 
Added check for version too. Thanks.